lxd/apparmor/rsync.go

```diff
@@ -0,0 +1,221 @@
+//go:build linux && cgo && !agent
+
+package apparmor
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+	"text/template"
+
+	"github.com/lxc/lxd/lxd/sys"
+	"github.com/lxc/lxd/shared"
+	"github.com/lxc/lxd/shared/subprocess"
+)
+
+var rsyncProfileTpl = template.Must(template.New("rsyncProfile").Parse(`#include <tunables/global>
+profile "{{ .name }}" flags=(attach_disconnected,mediate_deleted) {
+  #include <abstractions/base>
+
+  capability dac_override,
+  capability dac_read_search,
+  capability mknod,
+  capability chown,
+  capability fsetid,
+  capability fowner,
+
+  unix, # TODO Perhaps a stricter restriction can be made here
```

Comment:
Yeah, it's a bit broad. Could you check what it trips during an run with `flags=(complain)`, please?

Reply:
I have added `complain` and have received following log:
```
[Sat Apr 15 13:13:36 2023] audit: type=1400 audit(1681564416.406:56): apparmor="ALLOWED" operation="connect" profile="lxd_rsync-var-snap-lxd-common-lxd-storage-pools-default-containers-tc1" pid=3677 comm="lxd.debug" family="unix" sock_type="stream" protocol=0 requested_mask="send receive connect" denied_mask="send connect" addr=none peer_addr="@lxd/fb3fdc7e-4c97-4b00-b9e3-d920888d83ec" peer="unconfined"
```

I have fixed capability to make it more strict.

Thanks for suggestion.

Reply:
Thanks, with that information, I'd probably even try with:

```
unix (connect, send, receive) type=stream peer=(addr="@lxd/*"),
```

Reply:
Another thing probably worth testing is to interrupt an ongoing transfer. This is a situation where LXD would send a signal to rsync requiring an Apparmor rule along those lines (please test and adapt the signal set accordingly):

```
 signal (receive) set=(int,term) peer=(label=unconfined),
```